eden/scm/edenscm/ext/github/gh_submit.py

```diff
@@ -149,12 +149,13 @@ async def get_pull_request_details(
 
 
 def _parse_repository_from_dict(repo_obj, hostname: str, upstream=None) -> Repository:
+    default_branch = "main" if "defaultBranchRef" not in repo_obj else repo_obj["defaultBranchRef"]["name"]
```

Comment:
I think that we should be doing something like the following instead:

https://github.com/facebook/sapling/blob/6b7c5a62d1a60c4eb28e4f92e1f02c13f8c8f567/eden/scm/ghstack/github_utils.py#L91-L100

Though now that I add a pointer to that code, I notice that it needs to be updated to support GitHub Enterprise.